Gems/PhysX/Code/Editor/Source/Components/Conversion/PrefabConversionUtils.cpp

```diff
@@ -0,0 +1,253 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <AzCore/Asset/AssetManager.h>
+#include <AzFramework/Spawnable/Spawnable.h>
+#include <AzToolsFramework/API/EditorAssetSystemAPI.h>
+#include <AzToolsFramework/SourceControl/SourceControlAPI.h>
+
+#include <Editor/Source/Components/Conversion/PrefabConversionUtils.h>
+
+namespace PhysX::Utils
+{
+    static AZStd::optional<AZStd::string> GetFullSourceAssetPathById(AZ::Data::AssetId assetId)
+    {
+        AZStd::string assetPath;
+        AZ::Data::AssetCatalogRequestBus::BroadcastResult(assetPath, &AZ::Data::AssetCatalogRequests::GetAssetPathById, assetId);
+        AZ_Assert(!assetPath.empty(), "Asset Catalog returned an invalid path from an enumerated asset.");
+        if (assetPath.empty())
+        {
+            AZ_Warning(
+                "PhysXPrefabUtils",
+                false,
+                "Not able to get asset path for asset with id %s.",
+                assetId.ToString<AZStd::string>().c_str());
+            return AZStd::nullopt;
+        }
+
+        AZStd::string assetFullPath;
+        bool assetFullPathFound = false;
+        AzToolsFramework::AssetSystemRequestBus::BroadcastResult(
+            assetFullPathFound,
+            &AzToolsFramework::AssetSystem::AssetSystemRequest::GetFullSourcePathFromRelativeProductPath,
+            assetPath,
+            assetFullPath);
+        if (!assetFullPathFound)
+        {
+            AZ_Warning("PhysXPrefabUtils", false, "Source file of asset '%s' could not be found.", assetPath.c_str());
+            return AZStd::nullopt;
+        }
+
+        return { AZStd::move(assetFullPath) };
+    }
+
+    AZStd::vector<PrefabInfo> CollectPrefabs()
+    {
+        AZStd::vector<PrefabInfo> prefabs;
+
+        auto* prefabLoader = AZ::Interface<AzToolsFramework::Prefab::PrefabLoaderInterface>::Get();
+        auto* prefabSystemComponent = AZ::Interface<AzToolsFramework::Prefab::PrefabSystemComponentInterface>::Get();
```

Comment:
nit: Should add an AZ_Assert on the interface pointers 
```suggestion
        auto* prefabLoader = AZ::Interface<AzToolsFramework::Prefab::PrefabLoaderInterface>::Get();
        AZ_Assert(prefabLoader != nullptr, "Unable to get AzToolsFramework::Prefab::PrefabLoaderInterface interface");
        auto* prefabSystemComponent = AZ::Interface<AzToolsFramework::Prefab::PrefabSystemComponentInterface>::Get();
        AZ_Assert(prefabSystemComponent != nullptr, "Unable to get AzToolsFramework::Prefab::PrefabSystemComponentInterfaceinterface");
```